router/src/matching/horizontal/static_segment.rs

```diff
@@ -15,10 +15,20 @@ impl PossibleRouteMatch for () {
     fn generate_path(&self, _path: &mut Vec<PathSegment>) {}
 }
 
+pub trait Routable: Debug + Clone {
```

Comment:
For single-method traits, it's best to name it the same as the trait itself -- So given my proposal of `as_path`, I'd suggest `AsPath` for the trait name.

The `Debug + Clone` trait bounds are not necessary -- as long as they are implemented on your enum they do not need to be bounds on the trait.